### PR TITLE
Make choice of aggregator dynamic at session level and additional aggregators (FedOpt)

### PR DESCRIPTION
**Pull request by ahellander**
## Description

This pull request implements the foundation for more flexible experimentation with different optimization algorithms by making the aggregator used dynamically configurable on session level. Also introduces a scaffold for aggregators from the FedOpt family with an implementation of FedAdam.

New features: 

- Makes the configuration of aggregator dynamic per combiner per session (no longer set upon start of combiner server) 
- New aggregator plugin (fedopt)
- A new notebook showcasing how to use the client API for orchestrating experiments using different aggregators

Major changes: 
- Major refactor of the helpers. Both keras and pytorch examples now use list of numpy ndarray for model represenation/seralizations, and 'pytorchhelper' and 'kerashelper' have been consolidated into one 'numpyhelper'.
- The aggregator is no longer configurable upon starting up the combiner server, but needs to be set using a new RPC call SetAggregator.

### CODE REVIEW

File: examples/mnist-pytorch/client/fedn.yaml [hunk 1]
```diff
@@ -2,4 +2,5 @@ entry_points:
   train:
     command: /venv/bin/python entrypoint train $ENTRYPOINT_OPTS
   validate:
-    command: /venv/bin/python entrypoint validate $ENTRYPOINT_OPTS
\ No newline at end of file
+    command: /venv/bin/python entrypoint validate $ENTRYPOINT_OPTS
+helper: pytorchhelper
```
Comment (Wrede) [inline examples/mnist-pytorch/client/fedn.yaml:6]: what is this?

File: examples/notebooks/API_Example.ipynb
Comment (Wrede) [inline examples/notebooks/API_Example.ipynb:1]: I would add this to gitignore after it has been commited

File: fedn/fedn/network/combiner/aggregators/aggregatorbase.py [hunk 2]
```diff
@@ -81,36 +83,49 @@ def _validate_model_update(self, model_update):
...
-    def next_model_update(self, helper):
+    def next_model_update(self):
         """ Get the next model update from the queue.
 
         :param helper: A helper object.
         :type helper: object
-        :return: A tuple containing the model update, metadata and model id.
-        :rtype: tuple
+        :return: The model update.
+        :rtype: protobuf
```
Comment (Wrede) [inline fedn/fedn/network/combiner/aggregators/aggregatorbase.py:98]: protobuf?

File: fedn/fedn/network/combiner/combiner.py [hunk 3]
```diff
@@ -134,7 +134,7 @@ def __init__(self, config):
         self.server = Server(self, self.modelservice, grpc_config)
 
         # Set up round controller
-        self.control = RoundController(config['aggregator'], self.repository, self, self.modelservice)
+        self.control = RoundController(self.repository, self, self.modelservice)
```
Comment (Wrede) [inline fedn/fedn/network/combiner/combiner.py:137]: to avoid confusion with controller I think it should be called round_control

**Review by Wrede — CHANGES_REQUESTED** [review]
looks good! Some comments though

File: examples/mnist-pytorch/client/fedn.yaml (hunk 1 above)
Comment (ahellander) [reply examples/mnist-pytorch/client/fedn.yaml:6]: Ah, it sneakt in. I was thinking about a new way to specify the helper. Will remove for now.

File: fedn/fedn/network/combiner/aggregators/aggregatorbase.py (hunk 2 above)
Comment (ahellander) [reply fedn/fedn/network/combiner/aggregators/aggregatorbase.py:98]: Fixed

File: fedn/fedn/network/combiner/combiner.py (hunk 3 above)
Comment (ahellander) [reply fedn/fedn/network/combiner/combiner.py:137]: Agree, it is confusing. I changed it the RoundHandler (which is a more appropriate name)

File: examples/notebooks/API_Example.ipynb
Comment (ahellander) [reply examples/notebooks/API_Example.ipynb:1]: Done. I also moved it into the mnist-pytorch example

**Pull request merged**